### ISSUE TITLE
Rails resources, join tables, how to query?

### ISSUE THREAD

**Issue by tomBeach** [ga-wdi-exercises/project4#24]
I've tried five ways to query tables on either side of a join table but still can't seem to get rails magic happening.  It may be that my relationships are not correct.  The attached ERD is the design I am trying to use (no events or appointments tables yet).  I forgot to include the assignment table, which is a join between Person and Patient.  A patient will always be a person, but not the other way around.  (A person can be a doctor, nurse, etc.  Note that I am using two person ids in the patient table -- one to link to the patient's person data, and one to link to the patient's doctor's data.  The links happen via the appointments table -- not shown.  Is this asking for trouble?)
![erd](https://cloud.githubusercontent.com/assets/10157094/9905133/0aa25e74-5c51-11e5-870b-e746bb444d2f.jpg)
Here are my models:

```
class Person < ActiveRecord::Base
    has_many :assignments
    has_many :patients, through: :assignments
end
class Patient < ActiveRecord::Base
    has_many :assignments
end
class Assignment < ActiveRecord::Base
  belongs_to :person
  belongs_to :patient
end
```

Here is the controller:

```
    def index
        puts "******* WelcomeController.index ******* "
        @persons = Person.all
    end
```

Here is code from the generic view I am using to test query results.  It works but I try to use person.patient.room (using rails magic to link person and patient via the assignments join) it can't find "room".  

```
<h2>Patient Rooms</h2>

<div class="patients">
    <ul>
        <% @persons.each do |person| %>
            <% @patients = person.patients %>
            <% @patients.each do |patient| %>
                  <p><h1><%= patient.room %><%= patient.bed %><a href="/persons/<%= person.id %>"></h1><%= person.first_name %> <%= person.last_name %></a></p>
            <% end %>
        <% end %>
    </ul>
</div>
```

Basically I want to know if the models accurately represent the relationships I am trying to use...  Thanks!

**Comment by RobertAKARobin** [comment]
Are you getting an error?

**Comment by tomBeach** [comment]
Here is the code I was hoping to use (below).  It produces this error:

```
ActiveRecord::RecordNotFound in WelcomeController#index
Couldn't find Patient with 'id'=3
```

The reason it can't find patient with id=3 is that id=3 is a doctor.  3 is the id in the person table.  This implies the person.category = "patient" line is not working.  (I'm looking only for patients but need info from the persons table.)  Should I be using Assignment.fin instead of Patient.find?  

```
 <% @patient = Person.where(person.id, person.category = "patient") %>
      <% if @patient %>
          <% @patient_info = Patient.find(person.id) %>
          <p><h1><%= @patient_info.room %><%= @patient_info.bed %><a href="/persons/<%= person.id %>"></h1>
              <%= person.first_name %> <%= person.last_name %></a></p>
      <% end %>
```

**Comment by tomBeach** [comment]
Seed.rb fyi...

```
Person.delete_all
Patient.delete_all
Assignment.delete_all
Prescription.delete_all

# Person: "first_name", "last_name", "user_name", "password", "category"
Tom = Person.create(first_name: "Tom", last_name: "Beach", user_name: "tb", password: "tb12345", category: "nurse")
Tim = Person.create(first_name: "Tim", last_name: "Yoder", user_name: "ty", password: "ty12345", category: "patient")
Jim = Person.create(first_name: "Jim", last_name: "Jones", user_name: "jj", password: "jj12345", category: "doctor")
Joe = Person.create(first_name: "Joe", last_name: "Elixer", user_name: "je", password: "je12345", category: "patient")

# Assignment: "person_id", "patient_id", "category"
Assignment.create(person_id: 2, patient_id: 1, group: 2)
Assignment.create(person_id: 4, patient_id: 2, group: 2)

# Patient: "room", "bed", "diagnosis", "history"
pt_Tim = Patient.create(room: "218", bed: "A", diagnosis: "Left Knee Replacement", history: "this is a history")
pt_Joe = Patient.create(room: "218", bed: "B", diagnosis: "Right Knee Replacement", history: "this is Joe's history")

# Prescription: patient_id, person_id, generic_name, dose, route, frequency, side_effects, info_url
med1 = Prescription.create(patient_id: 1, person_id: 2, generic_name: "lisinopril", dose: "10mg", route: "pO", frequency: "qD", side_effects: "cough", info_url: "www.google.com")
med2 = Prescription.create(patient_id: 1, person_id: 2, generic_name: "aspirin", dose: "81mg", route: "pO", frequency: "qD", side_effects: "bleeding", info_url: "www.google.com")
med3 = Prescription.create(patient_id: 2, person_id: 4, generic_name: "propranolol", dose: "100mg", route: "pO", frequency: "q8h", side_effects: "headache", info_url: "www.google.com")
med4 = Prescription.create(patient_id: 2, person_id: 4, generic_name: "atorvastatin", dose: "60mg", route: "pO", frequency: "qHS", side_effects: "headache", info_url: "www.google.com")
```

**Comment by RobertAKARobin** [comment]
You just have the syntax a little wrong. It would be `category: "patient"`.

http://guides.rubyonrails.org/active_record_querying.html#hash-conditions

**Comment by tomBeach** [comment]
Same error after changing line:

```
<% @patient = Person.where(person.id, category = "patient") %>
```

I made a more accurate ERD with assignments table included, which brings up another question: how should I identify the doctor's id in the Patient table?  Should it be doctor_id (but referencing the doctor's person_id in Person table) or person_id.  There is potential confusion since doctors, nurses, patients and tech will all be in the Person table distinguished by category.  New patient model definition with doctor_id:

```
class Patient < ActiveRecord::Migration
    def change
        create_table :patients do |t|
            t.integer :doctor_id
            t.integer :room
            t.string :bed
            t.string :diagnosis
            t.text :history

            t.timestamps null: false
        end
    end
end
```

![erd](https://cloud.githubusercontent.com/assets/10157094/9906434/14319002-5c58-11e5-8a59-4128e5fe3f0e.png)

**Comment by tomBeach** [comment]
Ooops...  didn't get syntax change right...  fixed it but it's still not finding persons with a category of "patient".

```
<% @patient = Person.where(category: "patient") %>
```

**Comment by jshawl** [comment]
if you open up `$ rails c` - what is the output of `Person.where(category: "patient")`?
